### PR TITLE
Reset encoder from gui

### PR DESCRIPTION
**Pull request by ramensandwich**
Addresses issue #298

### CODE REVIEW

**Comment by ramensandwich** [comment]
Both low level and high level have been tested on the buggy.  Just need this to be checked off by @TrevorDecker and @VivaanBahl for correctness.

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java [hunk 1]
```diff
@@ -1,10 +1,13 @@
 package com.roboclub.robobuggy.ui;
 
 import com.roboclub.robobuggy.main.RobobuggyLogicNotification;
+import com.roboclub.robobuggy.nodes.sensors.RBSMNode;
 import com.roboclub.robobuggy.main.RobobuggyMessageLevel;
 import com.roboclub.robobuggy.messages.GuiLoggingButtonMessage;
 import com.roboclub.robobuggy.ros.NodeChannel;
 import com.roboclub.robobuggy.ros.Publisher;
+import com.roboclub.robobuggy.ros.Subscriber;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java]: Codacy Issue found: [Avoid unused imports such as 'com.roboclub.robobuggy.ros.Subscriber'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2219062423/issues/source?bid=2938665&fileBranchId=3172754#l9)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 2]
```diff
@@ -30,11 +30,12 @@
 	public static final int RBSM_COMMAND_PERIOD = 50;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Avoid excessively long variable names like RBSM_COMMAND_PERIOD](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2219062331/issues/source?bid=2938665&fileBranchId=3172754#l30)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java [hunk 3]
```diff
@@ -100,6 +116,20 @@ public double getPlaybackSpeed() {
...
 	}
+	
+	/**
+	 *  ResetEncoderButtonHandler
+	 * @author Sean
+	 * 
+	 * Sends a reset request message to low level when the "Reset Encoder" 
+	 * button on the gui is pressed
+	 *
+	 */
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java]: Codacy Issue found: [Each class should declare at least one constructor](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2219062423/issues/source?bid=2938665&fileBranchId=3172754#l128)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java [hunk 4]
```diff
@@ -1,10 +1,13 @@
 package com.roboclub.robobuggy.ui;
 
 import com.roboclub.robobuggy.main.RobobuggyLogicNotification;
+import com.roboclub.robobuggy.nodes.sensors.RBSMNode;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java]: Codacy Issue found: [Avoid unused imports such as 'com.roboclub.robobuggy.nodes.sensors.RBSMNode'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2219062423/issues/source?bid=2938665&fileBranchId=3172754#l4)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java [hunk 5]
```diff
@@ -23,6 +27,8 @@
  * {@link RobobuggyGUIContainer} used for logging information
  */
 public class LoggingPanel extends RobobuggyGUIContainer{
+	//TODO: test to make sure all these dont need to be static
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java]: Codacy Issue found: [Perhaps 'resetBtn' could be replaced by a local variable.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2219062423/issues/source?bid=2938665&fileBranchId=3172754#l31)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java [hunk 6]
```diff
@@ -299,6 +301,21 @@ public void actionPerformed(String topicName, Message m) {
...
 			});
+			new Subscriber(NodeChannel.ENCODER_RESET.getMsgPath(), new MessageListener() {
+				@Override
+				public void actionPerformed(String topicName, Message m) {
+					byte[] bytes = new byte[6];
+					bytes[0] = (byte)RBSerialMessage.getHeaderByte("RBSM_MID_ENC_RESET_REQUEST"); //Reset request header
+					bytes[1] = (byte)0;
+					bytes[2] = (byte)0;
+					bytes[3] = (byte)0;
+					bytes[4] = (byte)(0xc0);
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java]: why is this C0? I would prefer for us to not have magic values in the communication system

File: real_time/arduino_src/radio_buggy_mega/main.cpp [hunk 7]
```diff
@@ -313,6 +313,14 @@ int main(void)
                 // dipatch complete message
                 switch(new_command.message_id) 
                 {
+                    case RBSM_MID_ENC_RESET_REQUEST:
+                        cli();
+                        g_encoder_ticks = 0;
+                        sei();
+                        //Let high level know that the request went through
+                        g_rbsm.Send(RBSM_MID_ENC_RESET_CONFIRM, 1);
+                        printf("Encoder reset request received!\n");
```
Comment (VivaanBahl) [inline real_time/arduino_src/radio_buggy_mega/main.cpp]: should this be dbg_printf?

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java [hunk 8]
```diff
@@ -40,19 +47,26 @@
...
+		playBtn.setFont(new Font("serif", Font.PLAIN, 25));
 		playBtn.addActionListener(new PlayButtonHandler());
 		playBtn.setEnabled(true);
 		playBtn.setBackground(Color.BLUE);
+		
+		resetBtn = new JButton("RESET ENCODER");
+		resetBtn.setFont(new Font("seif", Font.PLAIN, 25));
+		resetBtn.addActionListener(new ResetEncoderButtonHandler());
+		resetBtn.setEnabled(true);
+		resetBtn.setBackground(Color.BLUE);
```
Comment (VivaanBahl) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java]: we should fix all these magic numbers

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java (hunk 1 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java]: Codacy Issue found: [Avoid unused imports such as 'com.roboclub.robobuggy.ros.Subscriber'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2230250215/issues/source?bid=2938665&fileBranchId=3172754#l9)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java (hunk 2 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Avoid excessively long variable names like RBSM_COMMAND_PERIOD](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2230250123/issues/source?bid=2938665&fileBranchId=3172754#l30)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java (hunk 3 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java]: Codacy Issue found: [Each class should declare at least one constructor](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2230250215/issues/source?bid=2938665&fileBranchId=3172754#l128)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java (hunk 4 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java]: Codacy Issue found: [Avoid unused imports such as 'com.roboclub.robobuggy.nodes.sensors.RBSMNode'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2230250215/issues/source?bid=2938665&fileBranchId=3172754#l4)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java (hunk 5 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java]: Codacy Issue found: [Perhaps 'resetBtn' could be replaced by a local variable.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2230250215/issues/source?bid=2938665&fileBranchId=3172754#l31)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 9]
```diff
@@ -1,5 +1,8 @@
 package com.roboclub.robobuggy.main;
 
+import com.roboclub.robobuggy.messages.EncoderResetMessage;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'com.roboclub.robobuggy.messages.EncoderResetMessage'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2230250125/issues/source?bid=2938665&fileBranchId=3172754#l3)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 10]
```diff
@@ -1,5 +1,8 @@
...
+import com.roboclub.robobuggy.ros.NodeChannel;
+import com.roboclub.robobuggy.ros.Publisher;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'com.roboclub.robobuggy.ros.Publisher'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2230250125/issues/source?bid=2938665&fileBranchId=3172754#l5)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java (hunk 9 above)
Comment (ramensandwich) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: This should not be included.

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java (hunk 2 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Avoid excessively long variable names like RBSM_COMMAND_PERIOD](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2231211611/issues/source?bid=2938665&fileBranchId=3172754#l30)

**Comment by VivaanBahl** [comment]
I am signing off on this

**Comment by TrevorDecker** [comment]
once this is tested on the buggy, I will be good for a merge

**Comment by ramensandwich** [comment]
This has been tested on the buggy.

**Comment by VivaanBahl** [comment]
^^ I am signing off and merging

**Pull request merged**